platform/kube/inject/initializer.go

```diff
@@ -14,72 +14,40 @@
 
 package inject
 
-// NOTE: This tool only exists because kubernetes does not support
-// dynamic/out-of-tree admission controller for transparent proxy
-// injection. This file should be removed as soon as a proper kubernetes
-// admission controller is written for istio.
-
 import (
 	"encoding/json"
-	"time"
 
 	"github.com/golang/glog"
 	appsv1beta1 "k8s.io/api/apps/v1beta1"
 	batchv1 "k8s.io/api/batch/v1"
+	v2alpha1 "k8s.io/api/batch/v2alpha1"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/api/extensions/v1beta1"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/strategicpatch"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
-
-	proxyconfig "istio.io/api/proxy/v1/config"
-	"istio.io/pilot/tools/version"
 )
 
 const (
 	initializerName = "sidecar.initializer.istio.io"
-	patchType       = types.StrategicMergePatchType
 )
 
 var ignoredNamespaces = []string{
-	"kube-system",
-	"kube-public",
+	"kube-system", // v1.NamespaceSystem
```

Comment:
maybe do the replacement?

Reply:
Yup. `NamespaceSystem` wasn't defined in `v1` along with `NamespaceDefault` so I thought it was a dep issue. Turns out some of these also appear in `metav1`.